libraries/core_libs/network/src/tarcap/packets_handlers/status_packet_handler.cpp

```diff
@@ -51,6 +52,17 @@ void StatusPacketHandler::process(const PacketData& packet_data, const std::shar
     auto const node_major_version = (*it++).toInt<unsigned>();
     auto const node_minor_version = (*it++).toInt<unsigned>();
     auto const node_patch_version = (*it++).toInt<unsigned>();
+    uint64_t node_history = (*it++).toInt<uint64_t>();
+
+    // If this is a light node and it cannot serve our sync request disconnect from it
+    if (node_history > 0) {
+      if (pbft_synced_period + node_history < peer_pbft_chain_size) {
+        LOG(log_nf_) << "Light node is not able to serve our syncing request. " << packet_data.from_node_id_.abridged()
+                     << " peer will be disconnected";
+        peers_state_->set_peer_light_node(peer->getId());
+        disconnect(peer->getId(), dev::p2p::UserReason);
```

Comment:
do we need to disconnect? I guess it is useless for us, but maybe we could still gossip transactions via RPC back to the network. I do not know what's right

Reply:
If we are out of sync I think we need to disconnect which will trigger us connecting to other nodes. I want to avoid the state where we are connected only to light nodes